src/components/Quiz/QuizWidget/QuizRadioGroup.tsx

```diff
@@ -147,64 +130,58 @@ const CustomRadio = ({
   })
 
   const buttonBg = useMemo<string>(() => {
-    if (!state.isChecked) return "background.highlight"
-    if (!isAnswerVisible) return "primary.base"
-    if (!isSelectedCorrect) return "error.base"
-    return "success.base"
+    if (!state.isChecked) return "bg-background-highlight"
+    if (!isAnswerVisible) return "bg-primary"
+    if (!isSelectedCorrect) return "bg-error"
+    return "bg-success"
   }, [state.isChecked, isAnswerVisible, isSelectedCorrect])
 
-  const primaryBaseColor = useToken("colors", "primary.base")
-
-  const getAnswerColor = (): ChakraProps["bg"] =>
-    isSelectedCorrect ? "success.base" : "error.base"
-
   const radioInputProps = getInputProps({ id: INPUT_ID })
 
   return (
     <>
-      <chakra.label
+      <label
         // `htmlFor` for proper accessibility with label and input
         {...getLabelProps({ htmlFor: INPUT_ID })}
-        cursor="pointer"
-        w="full"
+        className="w-full"
       >
         <HStack
-          data-testid="quiz-question-answer"
-          id={radioInputProps.value}
           {...getRadioProps()}
+          id={radioInputProps.value}
+          data-testid="quiz-question-answer"
+          data-group
+          data-answer-visible={isAnswerVisible || undefined}
```

Comment:
An alternative approach instead of rendering `data-*` attributes for dynamic styling would be dynamically assigned CSS vars at a parent level to be called when `hover` or `data-checked`